app/main.py

```
@@ -0,0 +1,525 @@
+    query_request: GraphQueryRequest,
+    neo4j_client: Neo4jClient = Depends(get_neo4j_client)
+):
+    """
+    Query the Neo4j knowledge graph.
+    """
+    try:
+        # Build the Cypher query
+        query = f"MATCH (n:{query_request.node_label})"
+        if query_request.filters:
+            query += " WHERE "
+            query += " AND ".join([f"n.{key} = ${key}" for key in query_request.filters.keys()])
```

Ruff (0.12.2)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)